File: src/RoboTasker.Domain/Robots/RobotPropertyValue.cs
using RoboTasker.Domain.Abstractions;

namespace RoboTasker.Domain.Robots;

public class RobotPropertyValue : TenantEntity
{
    public Robot Robot { get; set; } = null!;
    public Guid RobotId { get; set; }
    
    public RobotProperty Property { get; set; } = null!;
    public Guid PropertyId { get; set; }
    
    public string Value { get; set; } = null!;
}

File: src/Nerbotix.UI/src/app/models/robots/robots/robot-property-value.ts
import {TenantEntity} from '../../common/base-tenant-entity';
import {CategoryPropertyType} from '../categories/category-property-type';
import {Guid} from 'guid-typescript';

export class RobotPropertyValue extends TenantEntity {
  propertyId!: Guid;
  value!: any;
  type!: CategoryPropertyType;

  constructor(obj?: Partial<RobotPropertyValue>) {
    super(obj);
    if (obj) {
      Object.assign(this, obj);
    }
  }
}
